changelogs/fragments/334-mysql_user_fix_logic_on_oncreate.yml

```
@@ -0,0 +1,4 @@
+bugfixes:
+  - "mysql_user - fix logic when ``update_password`` is set to ``on_create`` for users using ``plugin*`` arguments (https://github.com/ansible-collections/community.mysql/issues/334)."
+    - ``on_create`` sets ``password`` to None for old mysql_native_authentication but not for authentiation methods which uses the ``plugin*`` arguments
+    - this PR changes this so ``on_create`` also exchange ``plugin``, ``plugin_hash_string``, ``plugin_auth_string`` to None in the list of arguments to change
```

```suggestion
  - "mysql_user - fix logic when ``update_password`` is set to ``on_create`` for users using ``plugin*`` arguments. The ``on_create`` value sets ``password`` to None for old mysql_native_authentication but not for authentiation methods which uses the ``plugin*`` arguments. This fix changes this so ``on_create`` also exchange ``plugin``, ``plugin_hash_string``, ``plugin_auth_string`` to None in the list of arguments to change (https://github.com/ansible-collections/community.mysql/issues/334)."
```
Thanks! A small formatting change suggestion to make our changelog entries consistent.

There's also the `DOCUMENTATION` block in the module's file, could you please describe what you wrote in the fragment in the corresponding option descriptions?